src/Console/SnapshotCommand.php

```diff
@@ -31,7 +31,7 @@ class SnapshotCommand extends Command
      */
     public function handle(Lock $lock, MetricsRepository $metrics)
     {
-        if ($lock->get('metrics:snapshot', 300)) {
+        if ($lock->get('metrics:snapshot', config('horizon.metrics.snapshot_lock', 300) - 30)) {
```

Comment:
The `-30` here is a slight change in behavior. We may not want to do that on a minor/patch release.

Reply:
That's the bugfix. Can be smaller too, but the point is:
1. Cron executes at 09:00:00
2. Command for metrics runs at 09:00:05 and is taken
3. Metric snapshot is taken
4. Cron executes at 09:05:00
5. Command for metrics runs at 09:05:01 and is not available, because lock is till 09:05:05.

That's the bug. The -30 is "a" solution, not the only one off-course. I'm open for suggestions.